### ISSUE TITLE
Wrong behavior of getPosition (attribute, attributePosition) in eslint-plugin-vue/lib/rules/attributes-order.js

### ISSUE THREAD

**Issue by tmcdos** [vuejs/eslint-plugin-vue#908]
**Tell us about your environment**	
* **ESLint version:**	5.16.0
* **eslint-plugin-vue version:**	5.2.2
* **Node version:**	10.5.0

**Please show your full configuration:**

```javascript
// https://eslint.org/docs/user-guide/configuring

module.exports = {
  root: true,
  parserOptions: {
    parser: 'babel-eslint'
  },
  env: {
    browser: true,
  },
  extends: [
    // https://github.com/vuejs/eslint-plugin-vue#priority-a-essential-error-prevention
    // consider switching to `plugin:vue/strongly-recommended` or `plugin:vue/recommended` for stricter rules.
    'plugin:vue/recommended', 
    // https://github.com/standard/standard/blob/master/docs/RULES-en.md
    '@vue/standard',
    'plugin:vue-i18n/recommended',
  ],
  settings: {
    'vue-i18n': {
      localeDir: './src/locale*.json' // extention is glob formatting!
    }
  },
  // add your custom rules here
  rules: {
    // allow async-await
    'generator-star-spacing': 'off',
    // allow debugger during development
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'semi': 'off',
    'eqeqeq': 'off',
    'brace-style': ['error', 'allman'],
    'object-curly-newline': ["error", {
        "ObjectExpression": { "multiline": true, "minProperties": 2, "consistent": true },
        "ObjectPattern": { "multiline": true },
        "ImportDeclaration": "never",
        "ExportDeclaration": { "multiline": true, "minProperties": 3 }
    }], 
    "object-property-newline": ["error", {"allowAllPropertiesOnSameLine": false}], 
    "comma-dangle": ["error", {
        "arrays": "only-multiline",
        "objects": "only-multiline",
        "imports": "never",
        "exports": "never",
        "functions": "never"
    }],
    'no-trailing-spaces': 'off',
    'quotes': 'off',
    'one-var': ['error', {'uninitialized': 'always', 'initialized': 'never'}],
    'no-v-html': 'off',
    "vue/max-attributes-per-line": ["error", {
      "singleline": 10,
      "multiline": {
        "max": 6,
        "allowFirstLine": true
      }
    }],
    "vue/order-in-components": ["error", 
      {
        "order": [
          "el",
          "name",
          "parent",
          "functional",
          ["delimiters", "comments"],
          ["components", "directives"],
          "extends",
          "mixins",
          "inheritAttrs",
          "model",
          ["props", "propsData"],
          "data",
          "filters",
          "computed",
          "watch",
          "LIFECYCLE_HOOKS",
          "methods",
          ["template", "render"],
          "renderError"
        ]
      }
    ],     
    'vue/singleline-html-element-content-newline': 'off',
    'vue/html-closing-bracket-spacing': 'off',
    'vue/no-v-html': 'off',
  }
} 
```

**What did you do?**	

```vue
<template>
  <v-layout v-if="totals && totals.unfiltered" class="domains_tabs">
    <v-btn v-for="(tab,idx) in valueTotals" :key="idx"
           :depressed="$route.query.status == tab.statusid"
           :color="$route.query.status == tab.statusid ? 'primary' : ''"
           :disabled="tab.statusid && tab.value == 0"
           @click="$emit('sheet',tab.statusid)"
    >
      {{ $t(tabLocalized[tab.statusid || '']) }}
      <v-badge v-if="tab.value" color="pink" class="ml-3 mr-3">
        <template slot="badge">
          {{ tab.value }}
        </template>
      </v-badge>
    </v-btn>
    <v-spacer />
    <v-menu left offset-y>
      <v-btn slot="activator" color="primary" style="padding-right: 8px;">
        {{ $t('domains.names.btn_export') }}
        <v-icon class="ml-2" style="font-size: 18px;">fa-arrow_down</v-icon>
      </v-btn>
      <v-list class="pa-0">
        <v-list-tile ripple @click="$emit('export','csv')">
          <v-list-tile-content>
            <v-list-tile-title>CSV</v-list-tile-title>
          </v-list-tile-content>
        </v-list-tile>
        <v-list-tile ripple @click="$emit('export','xls')">
          <v-list-tile-content>
            <v-list-tile-title>Excel</v-list-tile-title>
          </v-list-tile-content>
        </v-list-tile>
      </v-list>
    </v-menu>
  </v-layout>
</template>

<script>
export default
{
  name: 'DomainTabs',
  props:
    {
      totals:
        {
          type: Object,
          default: null
        }
    },
  computed:
    {
      tabLocalized ()
      {
        return {
          '': 'domains.names.tabs.all',
          '200': 'domains.names.tabs.active',
          '300': 'domains.names.tabs.transferred',
          '400': 'domains.names.tabs.inactive',
          '500': 'domains.names.tabs.expired',
          '10000': 'domains.names.tabs.archived',
        };
      },
      valueTotals ()
      {
        return this.totals ? Object.values(this.totals.unfiltered).sort((a, b) =>
        {
          return (a.statusid || 0) - (b.statusid || 0);
        }) : [];
      }
    },
}
</script>

<style lang="scss">
  .domains_tabs
  {
    .v-btn + .v-btn
    {
      margin-left: 0;
    }

    .theme--dark.v-btn.v-btn--disabled
    {
      color: rgba(0, 0, 0, 0.5) !important;
    }

    .theme--dark.v-btn.v-btn--disabled:not(.v-btn--icon):not(.v-btn--flat):not(.v-btn--outline)
    {
      background-color: rgba(0, 0, 0, 0.12) !important;
    }

    .v-badge__badge
    {
      width: auto;
      height: auto;
      min-width: 22px;
      min-height: 22px;
      padding: 1px 4px;
    }
  }

</style>
```

**What did you expect to happen?**
I expected to receive no errors and no warnings.

**What actually happened?**

```bash
Z:\2\2>node ../node_modules/eslint/bin/eslint --ext .js,.vue src/views/domains/list/modalflags.vue

Z:\2\2\src\views\domains\list\modalflags.vue
   3:57  warning  Attribute "@keydown.esc" should go before "scrollable"  vue/attributes-order
   3:93  warning  Attribute "@input" should go before "scrollable"        vue/attributes-order
  25:39  warning  Attribute "@click" should go before "dark"              vue/attributes-order
  26:37  warning  Attribute "@click" should go before "dark"              vue/attributes-order

? 4 problems (0 errors, 4 warnings)
  0 errors and 4 warnings potentially fixable with the `--fix` option.

Z:\2\2>
```
-----
The proposed solution looks like this (1 change in `getAttributeType` and 1 change in `getPosition`):
```javascript
function getAttributeType (name, isDirective) {
  if (typeof name === 'object') name = name.name; // <--- proposed change 1
  if (isDirective) {
    if (name === 'for') {
      return 'LIST_RENDERING'
    } else if (name === 'if' || name === 'else-if' || name === 'else' || name === 'show' || name === 'cloak') {
      return 'CONDITIONALS'
    } else if (name === 'pre' || name === 'once') {
      return 'RENDER_MODIFIERS'
    } else if (name === 'model') {
      return 'TWO_WAY_BINDING'
    } else if (name === 'on') {
      return 'EVENTS'
    } else if (name === 'html' || name === 'text') {
      return 'CONTENT'
    } else {
      return 'OTHER_DIRECTIVES'
    }
  } else {
    if (name === 'is') {
      return 'DEFINITION'
    } else if (name === 'id') {
      return 'GLOBAL'
    } else if (name === 'ref' || name === 'key' || name === 'slot' || name === 'slot-scope') {
      return 'UNIQUE'
    } else {
      return 'OTHER_ATTR'
    }
  }
}
function getPosition (attribute, attributePosition) {
  const attributeType = attribute.directive && (typeof attribute.key.name === 'object' ? attribute.key.name.name : attribute.key.name) === 'bind' // <--- proposed change 2
    ? getAttributeType(attribute.key.argument, false)
    : getAttributeType(attribute.key.name, attribute.directive)
  return attributePosition.hasOwnProperty(attributeType) ? attributePosition[attributeType] : -1
}
```

**Comment by vincekovacs** [comment]
The problem occurs with the following configuration:

```json
{
  "parser": "vue-eslint-parser",
  "parserOptions": {
    "parser": "@typescript-eslint/parser"
  },
  "extends": [
    "eslint:recommended",
    "plugin:vue/recommended",
    "typescript"
  ],
  "rules": {
    "@typescript-eslint/array-type": [
      "error",
      "array-simple"
    ],
    "@typescript-eslint/member-delimiter-style": [
      "error",
      {
        "multiline": {
          "delimiter": "none"
        }
      }
    ],
    "@typescript-eslint/no-explicit-any": false,
    "vue/html-indent": [
      "warn",
      4,
      {
        "baseIndent": 1
      }
    ],
    "vue/script-indent": [
      "warn",
      4,
      {
        "baseIndent": 1
      }
    ],
    "vue/max-attributes-per-line": false
  },
  "overrides": [
    {
      "files": "*.vue",
      "rules": {
        "@typescript-eslint/indent": false
      }
    }
  ],
  "env": {
    "amd": true
  }
}
```

And vue file:

```html
<div v-else-if="someCondition" key="some-key" class="row">
    <p>Lorem ipsum</p>
</div>

And the error message: `ESLint: Attribute "key" should go before "v-else-if". (vue/attributes-order)`

The above solution fix the problem.

**Comment by tmcdos** [comment]
Seems to be fixed in the master branch but not yet available in the official release (5.2.3)

**Comment by ota-meshi** [comment]
Thank you for this issue.

I close this issue since I have released the latest version 6.
